### DISCUSSION TITLE
CHMC:  A Family of MCMC Methods on Implicitly Defined Manifolds

### DISCUSSION THREAD

**Discussion by krzysztofrusek** [blackjax-devs/blackjax#642] [New features]
I am starting this thread is to discuss my work on the algorithm from [A Family of MCMC Methods on Implicitly Defined Manifolds](https://www.cs.toronto.edu/~mbrubake/projects/AISTATS12.pdf) by @mbrubake as discussed in in #538 .

In the PR I will migrate from [CHMC](https://github.com/krzysztofrusek/jax_chmc) to blackjax.

### TODO

- [x] RATTLE integrator
- [x] Implicit manifold metric and momentum generator
- [ ] `chmc` kernels
